bootstrap_cfn/r53.py

```diff
@@ -68,14 +69,71 @@ def update_dns_record(self, zone, record, record_type, record_value, is_alias=Fa
             changes.commit()
         return True
 
-    def get_record(self, zone, zone_id, record, record_type):
-        '''
-        '''
-        fqdn = "{0}.{1}.".format(record, zone)
-        rrsets = self.conn_r53.get_all_rrsets(zone_id, type=record_type, name=fqdn)
+    def delete_dns_record(self, zone_id, record_name, record_type, record_value, is_alias=False, dry_run=False):
+        """
+        Delete a dns record in route53
+        Args:
+            zone_id: a string specifying the zone id
+            record_name: a string for the record to update
+            record_value: a string if it is not an alias
+                            a list, if it is an alias, of parameters to pass to
+                            boto record.set_alias() function
+            record_type: a string to specify the record, eg "A"
+            is_alias:
+            dry_run:
+        Returns:
+             True if update successful or raises an exception if not
+        """
+        changes = boto.route53.record.ResourceRecordSets(self.conn_r53, zone_id)
+        change = changes.add_change("DELETE", record_name, record_type, ttl=60)
+        if is_alias:
+            # provide list of params as needed by function set_alias
+            # http://boto.readthedocs.org/en/latest/ref/route53.html#boto.route53.record.Record.set_alias
+            change.set_alias(*record_value)
+        else:
+            change.add_value(record_value)
+        if dry_run:
+            print(changes)
+        else:
+            res = changes.commit()
+        return res
+
+    def get_record(self, zone_name, zone_id, record_name, record_type):
+        """
+
+        Args:
+            zone_name:
+            zone_id:
+            record_name(String):
+            record_type:
+        Returns:
+            String or None, in the event of there being no A or TXT record
+        """
+        record_fqdn = "{0}.{1}.".format(record_name, zone_name)
+        rrsets = self.conn_r53.get_all_rrsets(zone_id, type=record_type, name=record_fqdn)
         for rr in rrsets:
-            if rr.type == record_type and rr.name == fqdn:
+            if rr.type == record_type and rr.name == record_fqdn:
                 if rr.type == 'TXT':
                     rr.resource_records[0] = rr.resource_records[0][1:-1]
+                if rr.type == 'A':
+                    return rr.alias_dns_name
```

Comment:
yes. But we wanted to keep returns as String type instead of "either String( `TXT` ) or RecordObejct ( `A` )".